src/classes/CRLP_Batch_Base_NonSkew.cls

```diff
@@ -323,4 +322,84 @@ public abstract class CRLP_Batch_Base_NonSkew extends CRLP_Batch_Base {
         options.detailObjType = this.detailObjectType;
         return options;
     }
+
+    /**
+     * @description Determines whether the rollup should be run in incremental mode
+     * @param jobType the type of rollup job
+     * @return true if the rollup should be run in incremental mode, otherwise false
+     */
+    private Boolean isIncrementalMode(CRLP_RollupProcessingOptions.RollupType jobType) {
+        //1. Only run incremental mode when initiated by the scheduled job
+        if (this.isScheduledJob == null || this.isScheduledJob == false) {
+            return false;
+        }
+
+        //2. Only run incremental mode when the custom settings are configured to enable incremental mode
+        Customizable_Rollup_Settings__c customizableRollupSettings = UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
+        if (jobType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit
+                && customizableRollupSettings.AccountHardCreditNonSkew_Incremental__c == true) {
+            return true;
+        } else if (jobType == CRLP_RollupProcessingOptions.RollupType.ContactHardCredit
+                && customizableRollupSettings.ContactHardCreditNonSkew_Incremental__c == true) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @description Produces the where clause that should be used for the inner join when in incremental mode
+     * @return The WHERE clause, as a string, that should be used to filter the detail records
+     */
+    private String getIncrementalModeWhereClause() {
+        String dateFieldNameForTesting = Test.isRunningTest() ? 'CloseDate' : 'SystemModstamp'; //Set to CloseDate for dev, will be set to SystemModstamp for testing/prod
```

Comment:
I'll need some help figuring out what to do here - I was thinking we might be able to test this better if we use an alternate date field for testing that would actually yield different results, since the SysModstamp will not actually filter down the Accounts in Test context.

Reply:
I'm wondering if we should be able to pass in a date to facilitate manual testing. Either way, I'd change `dateFieldNameForTesting` to `dateFieldName` since the variable is used in tests and in production.